Serialiser_Engine/Objects/BsonSerializers/MethodBaseSerializer.cs

```diff
@@ -166,10 +166,12 @@ private MethodBase GetMethod(string methodName, string typeName, List<string> pa
                     {
                         try
                         {
-                            if (methodName == ".ctor")
-                                method = type.GetConstructor(types.ToArray());
-                            else
-                                method = type.GetMethod(methodName, types.ToArray());
+                            Type[] typesArray = types.ToArray();
+                            if (typesArray != null)
+                                if (methodName == ".ctor")
+                                    method = type.GetConstructor(typesArray);
+                                else
+                                    method = type.GetMethod(methodName, typesArray);
```

Comment:
@alelom , can you add curly braces around the body of the `if` (line 170) for clarity ?